### PR TITLE
[FSDP] Add profiler to ImageNet

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request adds profiler to test/test_train_mp_imagenet_fsdp.py, and moves all the tracing part into the build_graph closure in test_train_mp_imagenet.py.

Test Plan:
CI.

### CODE REVIEW

File: test/test_train_mp_imagenet.py [hunk 1]
```diff
@@ -301,16 +301,17 @@ def train_loop_fn(loader, epoch):
...
-            _train_update, args=(device, step, loss, tracker, epoch, writer))
+          if FLAGS.ddp:
+            optimizer.step()
+          else:
+            xm.optimizer_step(optimizer)
+            tracker.add(FLAGS.batch_size)
+          if lr_scheduler:
+            lr_scheduler.step()
+          if step % FLAGS.log_steps == 0:
+            xm.add_step_closure(
```
Comment (JackCaoG) [inline test/test_train_mp_imagenet.py:312]: Let's move `step_ckousre` outside of `build_graph`, it is not actually building any graph. I felt like same way about the `lr_scheduler`. I think including `optimizer` make sense.

**Review by lsy323 — APPROVED** [review]

**Comment by lsy323** [comment]
LGTM!

File: test/test_train_mp_imagenet.py (hunk 1 above)
Comment (alanwaketan) [reply test/test_train_mp_imagenet.py:312]: Done.

**Comment by alanwaketan** [comment]
Thanks Siyuan for approving the change.

**Pull request merged**